protostar/starknet/new_arch/cheatable_cairo_entry_point.py

```diff
@@ -0,0 +1,281 @@
+        self,
+        state: State,
+        general_config: StarknetGeneralConfig,
+        resources_manager: Optional[ExecutionResourcesManager] = None,
+        tx_execution_context: Optional[TransactionExecutionContext] = None,
+    ) -> CallInfo:
+
+        new_config = deepcopy(general_config)
+        if self.max_steps is not None:
+
+            # Providing a negative value to Protostar results in infinite steps,
+            # this is here to mimic default Cairo behavior
+            value = None if self.max_steps < 0 else self.max_steps
+
+            # NOTE: We are doing it this way to avoid TypeError from typeguard
+            new_config.__dict__["invoke_tx_max_n_steps"] = value
+
+        return await super().execute_for_testing(
+            state=state,
+            general_config=new_config,
+            resources_manager=resources_manager,
+            tx_execution_context=tx_execution_context,
+        )
+
+    def execute(
+        self,
+        state: SyncState,
+        resources_manager: ExecutionResourcesManager,
+        general_config: StarknetGeneralConfig,
+        tx_execution_context: TransactionExecutionContext,
+    ) -> CallInfo:
+        new_config = deepcopy(general_config)
+        if self.max_steps is not None:
+
+            # Providing a negative value to Protostar results in infinite steps,
+            # this is here to mimic default Cairo behavior
+            value = None if self.max_steps < 0 else self.max_steps
+
+            # NOTE: We are doing it this way to avoid TypeError from typeguard
+            new_config.__dict__["invoke_tx_max_n_steps"] = value
+
+        if not isinstance(state, StateSyncifier):
+            sync_state = StateSyncifier(async_state=state, loop=get_running_loop())  # type: ignore
+        else:
+            sync_state = state
+
+        return super().execute(
+            state=sync_state,
+            resources_manager=resources_manager,
+            general_config=general_config,
+            tx_execution_context=tx_execution_context,
+        )
+
+    @classmethod
+    def create_with_cheaters(
+        cls,
+        contract_address: int,
+        calldata: List[int],
+        entry_point_selector: int,
+        cheaters: "CairoCheaters",
+    ) -> "CheatableCairoExecuteEntryPoint":
+        return cls(
+            entry_point_selector=entry_point_selector,
+            calldata=calldata,
+            contract_address=contract_address,
+            cheaters=cheaters,
+            code_address=None,
+            class_hash=None,
+            call_type=CallType.CALL,
+            entry_point_type=EntryPointType.EXTERNAL,
+            caller_address=0,
+        )
+
+    @classmethod
+    def factory(cls, cheaters: "CairoCheaters") -> type[ExecuteEntryPoint]:
+        def factory_function(*args: Any, **kwargs: Any) -> ExecuteEntryPoint:
+            return cls(*args, cheaters=cheaters, **kwargs)
+
+        return cast(type[ExecuteEntryPoint], factory_function)
```

Comment:
Don't we usually keep class methods at the of the class?

Reply:
done